### PR TITLE
Update our build system with obs-plugintemplate

### PR DESCRIPTION
**Pull request by umireon**
The build system on https://github.com/obsproject/obs-plugintemplate was updated.
It would be great to update our build system also.

### CODE REVIEW

**Comment by umireon** [comment]
I confirmed the notarization works.

File: buildspec.json [hunk 1]
```diff
@@ -1,86 +1,57 @@
...
-            "version": "28.0.1",
-            "repository": "https://github.com/obsproject/obs-studio.git",
-            "branch": "master",
-            "hash": "e8dc70d0eef4503378d6ac300e680215eb5c9379"
-        },
-        "prebuilt": {
-            "version": "2022-08-02",
-            "baseUrl": "https://github.com/obsproject/obs-deps/releases/download",
-            "label": "Pre-built obs-deps",
+            "version": "29.1.2",
```
Comment (royshil) [inline buildspec.json]: isnt this going to limit us to run on OBS v29.1.2 and up?
i think we should stick with v28 if possible

File: src/plugin-support.h [hunk 2]
```diff
@@ -1,6 +1,6 @@
 /*
-@PLUGIN_PRETTY_NAME@
-Copyright (C) 2021 @PLUGIN_AUTHOR@ @LINUX_MAINTAINER_EMAIL@
+OBS Background Removal Filter Plugin
+Copyright (C) 2021 Roy Shilkrot roy.shil@gmail.com
```
Comment (royshil) [inline src/plugin-support.h:3]: shuold we use the variables from buildpec,json

**Review by royshil — APPROVED** [review]
LGTM but check out a couple comments

File: buildspec.json (hunk 1 above)
Comment (umireon) [reply buildspec.json]: I forgot to fix this.

Comment (umireon) [reply buildspec.json]: Fixed

File: src/plugin-support.h (hunk 2 above)
Comment (umireon) [reply src/plugin-support.h:3]: Fixed

File: buildspec.json (hunk 1 above)
Comment (umireon) [reply buildspec.json]: It revealed that the new build system does not support OBS 28.0.1.

Comment (royshil) [reply buildspec.json]: i see it fails on mac because of SWIG
can we install SWIG to resolve?

Comment (umireon) [reply buildspec.json]: I have modified the build system and managed to be built on my local Mac with OBS 28.0.1.

Comment (umireon) [reply buildspec.json]: I conclude that the new build system requires huge modification if we go with OBS 28.0.1 and this is not reasonable.

Comment (umireon) [reply buildspec.json]: I tested the configuration of OBS 28.0.1 and so many problems other than SWIG occurred.

**Comment by royshil** [comment]
> I have to say that we have to bump OBS dependency to 29+ to adopt the new build system. The problem that occurs when we use OBS 28.0.1 is not so simple.

what do you suggest we do?
we update our support to 29.1+? 
many users are on 28, especially in Linux

**Comment by umireon** [comment]
@royshil Finally I have this plugin built with OBS 28.0.1.

**Review by royshil — APPROVED** [review]
alright looking good!

**Pull request merged**